label_studio/core/redis.py

```diff
@@ -122,28 +160,26 @@ def start_job_async_or_sync(job, *args, in_seconds=0, **kwargs):
 
     redis = redis_connected() and kwargs.get('redis', True)
     queue_name = kwargs.get('queue_name', 'default')
+
     if 'queue_name' in kwargs:
         del kwargs['queue_name']
     if 'redis' in kwargs:
         del kwargs['redis']
+
     job_timeout = None
     if 'job_timeout' in kwargs:
         job_timeout = kwargs['job_timeout']
         del kwargs['job_timeout']
+
     if redis:
-        # Auto-capture request_id from thread local and pass it via job meta
```

Comment:
It looks like this change breaks passing `request_id` into the job.

Reply:
It still does it, just using CurrentContext to handle all the thread_local related aspects. I had this solution on branch before this change for request_id, so I just merged it together. request_id is treated just as other job_data is, and is part of a dictionary of values that is set onto the job meta and restored from there on the job side.